### PR TITLE
v1.10 backports 2021-10-20

### PR DESCRIPTION
**Pull request by joamaki**
* #16507 -- workflows: Fix use of paths-filter on master pushes (@pchaigno)
 * #17511 -- node: Skip ipcache for remote node IPs if IPsec is enabled (@pchaigno)
 * #17157 -- contrib/backporting: Dockerize backporting scripts (@aditighag)
 * #17470 -- test/K8sVerifier: Cover several datapath configurations (@pchaigno)
 * #17573 --  bpf: Additional tail calls for IPvX-only setups (@pchaigno)

Edit (@kkourt):
Backporting #17470 showed that #17490 also affects v1.10, so I've also pulled in the fix: #17573

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 16507 17511 17157 17470 17573; do contrib/backporting/set-labels.py $pr done 1.10; done
```
or with
```
$ make add-label branch=v1.10 issues=16507,17511,17157,17470,17573
```

### CODE REVIEW

File: bpf/complexity-tests/419/bpf_host.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+-DSKIP_DEBUG=1 -DENABLE_IPV4=1 -DENABLE_IPV6=1 -DENABLE_HOST_SERVICES_TCP=1 -DENABLE_HOST_SERVICES_UDP=1 -DENABLE_HOST_REDIRECT=1 -DENABLE_ROUTING=1 -DNO_REDIRECT=1 -DPOLICY_VERDICT_NOTIFY=1 -DALLOW_ICMP_FRAG_NEEDED=1 -DENABLE_IDENTITY_MARK=1 -DMONITOR_AGGREGATION=3 -DCT_REPORT_FLAGS=0x0002 -DENABLE_HOST_FIREWALL=1 -DHAVE_LPM_TRIE_MAP_TYPE=1 -DHAVE_LRU_HASH_MAP_TYPE=1 -DENABLE_MASQUERADE=1 -DENABLE_SRC_RANGE_CHECK=1 -DENABLE_NODEPORT=1 -DENABLE_NODEPORT_ACCELERATION=1 -DENABLE_SESSION_AFFINITY=1 -DENABLE_DSR_ICMP_ERRORS=1 -DENABLE_DSR=1 -DENABLE_DSR_HYBRID=1 -DENABLE_IPV4_FRAGMENTS=1 -DENCAP_IFINDEX=1 -DTUNNEL_MODE=1 -DENABLE_EGRESS_GATEWAY=1
```
Comment (pchaigno) [inline bpf/complexity-tests/419/bpf_host.txt]: A couple config macros were [removed in master](https://github.com/cilium/cilium/commit/040d264ebcd7b9fb1ce60baeb2beffe1c89576c9) but are still present in v1.10, so we need to update those configurations to take those macros into account. I think simply replacing the substring `-DENABLE_NODEPORT=1 -DENABLE_NODEPORT_ACCELERATION=1 -DENABLE_SESSION_AFFINITY=1 -DENABLE_DSR_ICMP_ERRORS=1 -DENABLE_DSR=1 -DENABLE_DSR_HYBRID=1` in all `bpf/complexity-tests/` files with `-DENABLE_NODEPORT=1 -DENABLE_EXTERNAL_IP=1 -DENABLE_NODEPORT_ACCELERATION=1 -DENABLE_SESSION_AFFINITY=1 -DENABLE_DSR_ICMP_ERRORS=1 -DENABLE_DSR=1 -DENABLE_HOSTPORT=1 -DENABLE_LOADBALANCER -DENABLE_DSR_HYBRID=1` should do it. Could you make that change? :pray:

**Review by pchaigno — CHANGES_REQUESTED** [review]
Small change required below—small in scope/time but maybe not in line changes :sweat_smile:

**Review by aditighag — APPROVED** [review]
Backport of my commit looks good. Thanks!

File: bpf/complexity-tests/419/bpf_host.txt (hunk 1 above)
Comment (joamaki) [reply bpf/complexity-tests/419/bpf_host.txt]: Heading to PTO, so I'll hand this over to @kkourt who volunteered to help out :smile:

Comment (pchaigno) [reply bpf/complexity-tests/419/bpf_host.txt]: @joamaki Thanks for the heads up. Enjoy :mount_fuji: !

@kkourt It should be a simple sed command to replace :slightly_smiling_face: Happy to help if anything goes wrong of course!

**Review by aanm — APPROVED** [review]

File: bpf/complexity-tests/419/bpf_host.txt (hunk 1 above)
Comment (kkourt) [reply bpf/complexity-tests/419/bpf_host.txt]: @pchaigno I did as suggested. Could you please have another look?

**Review by pchaigno — APPROVED** [review]
Thanks!

**Comment by pchaigno** [comment]
/test-backport-1.10

**Comment by kkourt** [comment]
@pchaigno, I guess it is  [expected](https://github.com/cilium/cilium/pull/17573#issuecomment-940972690) that the verifier test will fail on 4.19 as it does?

**Review by pchaigno — APPROVED** [review]
>  Seems to me that it's better to pick up the fix now rather than on the next round.

Definitely, otherwise it would break the tests.

**Pull request merged**